### DISCUSSION TITLE
DRO for interface

### DISCUSSION THREAD

**Discussion by Armandleo93** [winder/Universal-G-Code-Sender#1863] [Help]
Hi, 

Not sure where to post this so I thought I'd start here. 

I'm trying to build and interface for my cnc, one similar to the openbuilds interface. 

I'm just trying to work out of there is a way of getting info from the DRO out of UGS into something like a raspberry pi pico and display it on a screen all connected via USB?

Thanks for any suggestions

**Comment by chrisla23** [comment]
Not up to the second, but I pull it out with the pendant API and send it to
an LCD.

Eg.

FOR /F "delims=" %%A IN ('xidel.exe
http://192.168.7.85:8080/api/v1/status/getStatus -e
"rowCount:=($json).rowCount" -e
"completedRowCount:=($json).completedRowCount" -e
"remainingRowCount:=($json).remainingRowCount" -e
"sendDuration:=($json).sendDuration" -e
"sendRemainingDuration:=($json).sendRemainingDuration"  -e
"workCoordX:=round(($json).workCoord.x,2)" -e
"workCoordY:=round(($json).workCoord.y,2)" -e
"workCoordZ:=round(($json).workCoord.z,2)" -e "state:=($json).state" -e
"fileName:=($json).fileName" --output-format=cmd') DO %%A

The tool I am using to extract the JSON is cross-platform.

-Chris

**Reply by Armandleo93** [reply]
Thanks for your quick response Chrisla23! 

Okay, i suppose that if you can pull info from the API in that way it could potentially take info as well from some push buttons right? 

The thought is to make a pendant that connects over usb to a raspberry pi running UGS that has some info displaying on a 3 or 4" lcd screen. Would this be doable as you've suggested above?

Thanks again!

**Comment by chrisla23** [comment]
Correct.

You may want to take a look at this project:  https://github.com/aleslukek/UGS-Wifi-Pendant

Here is an example of sending some gcode: 

curl -H "Content-Type: application/json" -X POST -d "{\"commands\":\"G20G90G53G0Z0;G53G0X0Y0\"}" http://192.168.7.85:8080/api/v1/machine/sendGcode

I would say you are better off connecting to UGS over wifi than USB. 

-Chris

**Reply by Armandleo93** [reply]
Ah that project is almost exactly what I was looking for! Thanks so much!
